indexing-hadoop/src/main/java/io/druid/indexer/HadoopIngestionSpec.java

```diff
@@ -116,25 +122,40 @@ public static HadoopIngestionSpec updateSegmentListIfDatasourcePathSpecIsUsed(
 
     Map<String, Object> pathSpec = spec.getIOConfig().getPathSpec();
     Map<String, Object> datasourcePathSpec = null;
-    if(pathSpec.get(type).equals(dataSource)) {
+    if (pathSpec.get(type).equals(dataSource)) {
       datasourcePathSpec = pathSpec;
-    } else if(pathSpec.get(type).equals(multi)) {
+    } else if (pathSpec.get(type).equals(multi)) {
       List<Map<String, Object>> childPathSpecs = (List<Map<String, Object>>) pathSpec.get(children);
-      for(Map<String, Object> childPathSpec : childPathSpecs) {
+      for (Map<String, Object> childPathSpec : childPathSpecs) {
         if (childPathSpec.get(type).equals(dataSource)) {
           datasourcePathSpec = childPathSpec;
           break;
         }
       }
     }
+
     if (datasourcePathSpec != null) {
       Map<String, Object> ingestionSpecMap = (Map<String, Object>) datasourcePathSpec.get(ingestionSpec);
-      DatasourceIngestionSpec ingestionSpecObj = jsonMapper.convertValue(ingestionSpecMap, DatasourceIngestionSpec.class);
+      DatasourceIngestionSpec ingestionSpecObj = jsonMapper.convertValue(
+          ingestionSpecMap,
+          DatasourceIngestionSpec.class
+      );
       List<DataSegment> segmentsList = segmentLister.getUsedSegmentsForInterval(
           ingestionSpecObj.getDataSource(),
           ingestionSpecObj.getInterval()
       );
-      datasourcePathSpec.put(segments, segmentsList);
+      VersionedIntervalTimeline<String, DataSegment> timeline = new VersionedIntervalTimeline<>(Ordering.natural());
```

Comment:
@gianm , I was wondering why we need to scan the timeline here since the returned segments from ```segmentLister.getUsedSegmentsForInterval``` already did that?